shell/shared/shell.js

```diff
@@ -689,6 +689,51 @@ if (Meteor.isClient) {
     },
     accountSettingsUi: function () {
       return makeAccountSettingsUi();
+    },
+    firstTimeBillingPromptState: function () {
+      // Should we show the first-time billing plan selector?
+
+      // Don't show if billing is not enabled.
+      if (!window.BlackrockPayments) return;
+
+      var user = Meteor.user();
+
+      // Don't show if not logged in.
+      if (!user) return;
+
+      // Don't show if the user has selected a plan already.
+      if (user.plan && !Session.get("firstTimeBillingPromptOpen")) return;
+
+      // Only show to account users (not identities).
+      if (!user.loginIdentities) return;
+
+      // Don't show to demo users.
+      if (user.expires) return;
+
+      // Don't show when viewing another user's grain. We don't want to scare people away from
```

Comment:
For grain routes, the code below checks whether the active grain is owned by the current user, and does not display the prompt if it isn't.